File: src/main/kotlin/ca/myasir/linksqueeze/model/UrlMetric.kt
package ca.myasir.linksqueeze.model

import ca.myasir.linksqueeze.dao.MetricType
import ca.myasir.linksqueeze.util.UrlHash
import ca.myasir.linksqueeze.util.UserId
import java.time.ZonedDateTime

data class UrlMetric(
    val urlHash: UrlHash,
    val metricType: MetricType,
    val count: Double,
    val date: ZonedDateTime
)

File: link-squeeze-website/src/model/UrlMetric.ts
import {Type} from "class-transformer"

export class UrlMetric {

  urlHash: string
  metricType: string
  count: number

  @Type(() => Date)
  date: Date

  constructor(urlHash: string, metricType: string, count: number, date: Date) {
    this.urlHash = urlHash;
    this.metricType = metricType;
    this.count = count;
    this.date = date;
  }
}
